### PR TITLE
KV Key Whitespace Warning

### LINKED ISSUE #6714: Web UI does not trim leading/trailing whitespaces in a secret's key
**Describe the bug**
The Web UI does not trim leading/trailing whitespaces in a secret's key. Consequently, there might be issues when accessing the key. These leading/trailing whitespaces are hard to detect in the UI, especially since the input fields have a custom style and padding.

**To Reproduce**
1. Create a secret using the Web UI.
2. Add a key-value pair including leading/trailing whitespaces in the key.
3. Try to access the value via the API using the key without any leading/trailing whitespaces. It cannot be found.

**Expected behavior**
Trim leading/trailing whitespaces in keys. If this is not desired, consider to alert the user in the Web UI in such cases.
Also consider to alert the user if there are leading/trailing whitespaces in the corresponding value. However, trimming the value may not be beneficial (in general, modifying/normalizing passwords automatically is a bad idea).

**Environment:**
* Vault Version: 0.10.3

### PR DESCRIPTION
**Pull request by zofskeez**
This PR adds a warning when whitespace is detected in a key of a KV engine secret.

image
image
image

Closes #6714

### CODE REVIEW

File: ui/lib/core/addon/components/kv-object-editor.js [hunk 1]
```diff
@@ -81,4 +84,16 @@ export default class KvObjectEditor extends Component {
       this.args.onKeyUp(event.target.value);
     }
   }
+  @action
+  validateKey(rowIndex, event) {
+    const { value } = event.target;
+    const keyHasWhitespace = new RegExp('\\s', 'g').test(value);
```
Comment (hellobontempo) [inline ui/lib/core/addon/components/kv-object-editor.js:90]: We have a util you could use here https://github.com/hashicorp/vault/blob/main/ui/app/utils/validators.js#L31-L34

Comment (zofskeez) [reply ui/lib/core/addon/components/kv-object-editor.js:90]: That lives in the main app unfortunately. I thought about moving it to the core addon but given that it's a one liner I opted to go this route.

Comment (hellobontempo) [reply ui/lib/core/addon/components/kv-object-editor.js:90]: Ah - I forgot that the models that use it also live in the main app 🙃

**Review by hellobontempo — APPROVED** [review]
Thanks for adding a test!

**Pull request merged**